cube/cube.cpp

```diff
@@ -774,111 +670,101 @@ void Demo::draw() {
-                              .setClearValueCount(2)
-                              .setPClearValues(clearValues);
-
-    auto result = commandBuffer.begin(&commandInfo);
+    auto result = commandBuffer.begin(vk::CommandBufferBeginInfo().setFlags(vk::CommandBufferUsageFlagBits::eSimultaneousUse));
     VERIFY(result == vk::Result::eSuccess);
 
-    commandBuffer.beginRenderPass(&passInfo, vk::SubpassContents::eInline);
+    commandBuffer.beginRenderPass(vk::RenderPassBeginInfo()
+                                      .setRenderPass(render_pass)
+                                      .setFramebuffer(swapchain_image_resource.framebuffer)
+                                      .setRenderArea(vk::Rect2D(vk::Offset2D(0, 0), vk::Extent2D(static_cast<uint32_t>(width),
+                                                                                                 static_cast<uint32_t>(height))))
+                                      .setClearValueCount(2)
+                                      .setPClearValues(clearValues),
+                                  vk::SubpassContents::eInline);
+
     commandBuffer.bindPipeline(vk::PipelineBindPoint::eGraphics, pipeline);
-    commandBuffer.bindDescriptorSets(vk::PipelineBindPoint::eGraphics, pipeline_layout, 0, 1,
-                                     &swapchain_image_resources[current_buffer].descriptor_set, 0, nullptr);
+    commandBuffer.bindDescriptorSets(vk::PipelineBindPoint::eGraphics, pipeline_layout, 0, swapchain_image_resource.descriptor_set,
+                                     nullptr);
     float viewport_dimension;
     float viewport_x = 0.0f;
     float viewport_y = 0.0f;
     if (width < height) {
-        viewport_dimension = (float)width;
+        viewport_dimension = static_cast<float>(width);
         viewport_y = (height - width) / 2.0f;
     } else {
-        viewport_dimension = (float)height;
+        viewport_dimension = static_cast<float>(height);
         viewport_x = (width - height) / 2.0f;
     }
-    auto const viewport = vk::Viewport()
-                              .setX(viewport_x)
-                              .setY(viewport_y)
-                              .setWidth((float)viewport_dimension)
-                              .setHeight((float)viewport_dimension)
-                              .setMinDepth((float)0.0f)
-                              .setMaxDepth((float)1.0f);
-    commandBuffer.setViewport(0, 1, &viewport);
-
-    vk::Rect2D const scissor(vk::Offset2D(0, 0), vk::Extent2D(width, height));
-    commandBuffer.setScissor(0, 1, &scissor);
+
+    commandBuffer.setViewport(0, vk::Viewport()
+                                     .setX(viewport_x)
+                                     .setY(viewport_y)
+                                     .setWidth(static_cast<float>(viewport_dimension))
+                                     .setHeight(static_cast<float>(viewport_dimension))
+                                     .setMinDepth(static_cast<float>(0.0f))
```

Comment:
Casting `float` to `float`?

Reply:
The old code did `(float)value` here a bunch, was just robotically moving to static_cast. Turns out its entirely unnecessary!